### PR TITLE
Hide exact agent state to everyone except the agent

### PR DESCRIPTION
**Pull request by Razvan00Rusu**
Refactor state view to obscure exact values of agent health and stamina, instead offering a range. The way it currently works means that agents wont even know their own exact HP/Stamina. This probably makes sense conceptually as agents would probably know something like 'I'm Fine', 'It hurts a little', 'it hurts a lot', 'im literally almost dead' but not 'I'm 24.7% dead'.

Up for discussion.

### CODE REVIEW

**Comment by 0x6770** [comment]
fixed by rebase to main and remove that line

File: pkg/infra/game/state/view.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+func (v *View) MonsterHealth() uint {
+	return v.monsterHealth
+}
+
+func (v *View) MonsterAttack() uint {
+	return v.monsterAttack
+}
+
+func (v *View) AgentState() *immutable.Map[commons.ID, HiddenAgentState] {
+	return v.agentState
```
Comment (neeldug) [inline pkg/infra/game/state/view.go:58]:
```suggestion
func (v *View) AgentState() immutable.Map[commons.ID, HiddenAgentState] {
	return *v.agentState
```

File: pkg/infra/game/state/view.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+func (s *State) ToView() *View {
+	b := immutable.NewMapBuilder[commons.ID, HiddenAgentState](nil)
+	for uuid, state := range s.AgentState {
```
Comment (neeldug) [inline pkg/infra/game/state/view.go:63]: Do agents get a hidden view of their own state?

**Review by neeldug — COMMENTED** [review]
Looks pretty good to me so far.

**Comment by Razvan00Rusu** [comment]
Ok so based on the conversation we had in the meeting today, we want agents to have a discrete view of other agent's HP and stamina but a more complete view of its own values. 

Base/Bonus Attack and Shield should be public

Inventory should be private (but not implemented atm).

**Review by neeldug — APPROVED** [review]
LGTM

**Pull request merged**